client/state/user-devices/reducer.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	USER_DEVICES_ADD,
+} from 'state/action-types';
+
+export default createReducer( {}, {
+	[ USER_DEVICES_ADD ]: ( state, { devices } ) => devices
```

Comment:
seems fine either way. with 300 sites or so my two devices took up over 100 KB of data to store so that's definitely something to consider.